src/pal/src/thread/process.cpp

```diff
@@ -1518,6 +1518,9 @@ class PAL_RuntimeStartupHelper
             case EEXIST:
                 pe = ERROR_ALREADY_EXISTS;
                 break;
+            case ENOSPC:
```

Comment:
As we have seen in the sem issue that hit our OSX machines, this error doesn't mean the disk is full, so the ERROR_DISK_FULL code seems misleading. 
I would recommend using ERROR_TOO_MANY_SEMAPHORES here (you need to add it to pal_error.h - its value is 100)